mojo/edk/embedder/tcp_platform_handle_utils_posix.cc

```diff
@@ -44,8 +44,22 @@ ScopedPlatformHandle CreateTCPSocket(bool needs_connection, int protocol) {
 ScopedPlatformHandle CreateTCPClientHandle(size_t port) {
   std::string server_address = "127.0.0.1";
   base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  if (command_line->HasSwitch(switches::kServerAddress))
-    server_address = command_line->GetSwitchValueASCII(switches::kServerAddress);
+  if (command_line->HasSwitch(switches::kServerAddress)) {
+    server_address =
+        command_line->GetSwitchValueASCII(switches::kServerAddress);
+    int status;
+    struct addrinfo* result = NULL;
+    status = getaddrinfo(server_address.c_str(), NULL, NULL, &result);
+    if (status == 0) {
+      char hostname[NI_MAXHOST] = "";
+      status = getnameinfo(result->ai_addr, result->ai_addrlen, hostname,
+                           NI_MAXHOST, NULL, 0, NI_NUMERICHOST);
+      if (status == 0 && *hostname) {
+        server_address = std::string(hostname);
```

Comment:
My question may be naive. We are updating server_address to host name. But later we use inet_addr(server_address).
As per https://linux.die.net/man/3/inet_addr, it states inet_addr accepts numbers-and-dots notation as input.
Should that be ok ?